src/intrprtr.c

```diff
@@ -1937,20 +1876,19 @@ void            IntrCharExpr (
 */
 static Obj GetFromStack(Obj cycle, Int j)
 {
-    return PopObj();
+    IntrState * intr = (IntrState *)cycle;
+    return PopObj(intr);
```

Comment:
Another "real" change: we need to pass `intr` to `PopObj`. Luckily, `ScanPermCycle` (which calls `GetFromStack` as a callback) already allows us to pass in a pointer of our choice, namely `cycle`, so we abuse that to pass in `intr`.